hooks/post_gen_project.py

```diff
@@ -0,0 +1,66 @@
+"""Post project generation hook."""
+
+import subprocess
+import sys
+
+_EXIT_FAILURE = 1
+_EXIT_SUCCESS = 0
+
+
+def main(initialise_git_repository: str) -> int:
+    """
+    Create a git repository on generation of the project.
+
+    Args:
+        initialise: Whether to initialise the repo
+
+    Returns:
+        The return code of the process
+    """
+    if initialise_git_repository == "True":
+        # initialise git repo
+        try:
+            subprocess.run(["git", "init"], check=True)  # noqa: S603,S607
+            # old versions of git still default to `master`
+            subprocess.run(
+                ["git", "branch", "-M", "main"],  # noqa: S603,S607
+                check=True,
+                capture_output=True,
+            )
+        except FileNotFoundError:
+            print("git isn't installed")  # noqa: T201
+            return _EXIT_FAILURE
+        except subprocess.CalledProcessError as e:
+            print(  # noqa: T201
+                f"There was an error with git: {e.returncode}\n{e.stderr}"
+            )
+            return _EXIT_FAILURE
+
+        # check for presence of GitHub CLI
+        try:
+            subprocess.run(["gh", "--version"], check=True, capture_output=True)  # noqa: S603,S607
+            print(  # noqa: T201
+                "GitHub CLI detected, you can create a repo with the following:\n\n"
+                "gh repo create {{cookiecutter.project_slug}} -d "
+                "'{{cookiecutter.project_short_description}}' --public -r "
+                "origin --source {{cookiecutter.project_slug}}"
+            )
```

Comment:
```suggestion
            try:
                subprocess.run(shlex.split("gh repo view {{cookiecutter.project_slug}} --json id -q '.id'"), check=True, capture_output=True) 
                print(  # noqa: T201
                    "GitHub CLI detected, but {{cookiecutter.project_slug}} already exists. You can create a repo with the following:\n\n"
                    "gh repo create {{cookiecutter.project_slug}}-new -d "
                    "'{{cookiecutter.project_short_description}}' --public -r "
                    "origin --source {{cookiecutter.project_slug}}"
                )           
            except subprocess.CalledProcessError as e:
                # fails because the repository doesn't exist
                subprocess.run(shlex.split("gh repo create {{cookiecutter.project_slug}} -d '{{cookiecutter.project_short_description}}' --public -r origin --source {{cookiecutter.project_slug}}"), check=True, capture_output=True)                 
```

Reply:
Great idea. I can't click `commit suggestion` from some reason, so will do manually

Reply:
Reverting as this is getting too complex and doesn't work as expected